flink-connector-examples/src/main/java/io/pravega/example/flink/watermark/reader/PravegaWatermarkIngestion.java

```diff
@@ -0,0 +1,98 @@
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.connectors.flink.PravegaConfig;
+import io.pravega.example.flink.watermark.Constants;
+import io.pravega.example.flink.watermark.SensorData;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/*
+ * Parameters
+ *     -controller tcp://<controller_host>:<port>, e.g., tcp://localhost:9090
+ *
+ * Functionality
+ *     - Ingest simulated sine-wave sensor data with event timestamp into a Pravega stream(watermark-examples/input).
+ */
+public class PravegaWatermarkIngestion {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PravegaWatermarkIngestion.class);
+
+    private static final double EVENT_VALUE_INCREMENT = 0.01; // Should be < 1
+    private static final int WRITER_SLEEP_MS = 100;
+
+    public static void main(String[] args) throws InterruptedException {
+        ParameterTool params = ParameterTool.fromArgs(args);
+        // The writer will contact with the Pravega controller to get information about segments.
+        URI pravegaControllerURI = URI.create(params.get(Constants.CONTROLLER_ADDRESS_PARAM, Constants.CONTROLLER_ADDRESS));
+        final int numEvents = Constants.EVENTS_NUMBER;
+
+        // StreamManager helps us to easily manage streams and copes.
+        StreamManager streamManager = StreamManager.create(pravegaControllerURI);
+
+        // A scope is a namespace that will be used to group streams (e.g., like dirs and files).
+        streamManager.createScope(Constants.DEFAULT_SCOPE);
+
+        PravegaConfig pravegaConfig = PravegaConfig.fromDefaults()
+                .withControllerURI(pravegaControllerURI)
+                .withDefaultScope(Constants.DEFAULT_SCOPE);
+
+        // Here we configure the new stream (e.g. scaling policy, retention policy).
+        StreamConfiguration streamConfiguration = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(Constants.PARALLELISM))
+                .build();
+
+        // Create a Pravega stream to write data (if it does not exist yet).
+        streamManager.createStream(Constants.DEFAULT_SCOPE, Constants.INPUT_STREAM, streamConfiguration);
+
+        // Create the client factory to instantiate writers and readers.
+        try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(Constants.DEFAULT_SCOPE, pravegaConfig.getClientConfig())) {
+
+            // Create a writer to write events in the stream.
+            EventStreamWriter<SensorData> writer = clientFactory.createEventWriter("watermarking-events",
+                    Constants.INPUT_STREAM,
+                    new JavaSerializer<>(), EventWriterConfig.builder().build());
+
+            for (double i = 1; i <= numEvents; i++) {
+                // Write an event for each sensor.
```

Comment:
The number of sensors needn't match the parallelism of task; it is confusing to suggest it by using the `PARALLELISM` constant.   Suggest another constant.

Reply:
Fixed